prime-router/src/main/kotlin/azure/CheckFunction.kt

```diff
@@ -301,83 +309,150 @@ class CheckFunction : Logging {
             responseBody.add("**** ${receiver.fullName}:  no transport defined.")
             return false
         }
+        responseBody.add("**** Receiver Status: ${receiver.customerStatus}")
+        val receiverConnected = when (receiver.transport) {
+            is SFTPTransportType -> testSftp(receiver.transport, receiver, sftpFile, responseBody)
+            is RESTTransportType -> testRest(receiver.transport, receiver, responseBody)
+            // todo add other types of transports as needed.
+            else -> {
+                responseBody.add(
+                    "**** ${receiver.fullName}: No test implemented for transport type ${receiver.transport.type}"
+                )
+                false
+            }
+        }
+        return if (receiverConnected) {
+            responseBody.add("**** ${receiver.fullName}: OK")
+        } else {
+            responseBody.add("**** ${receiver.fullName}: FAILED")
+            return false
+        }
+    }
+
+    /**
+     * Check if the REST transport can connect to the receiver
+     * and authenticate with the credentials in the vault
+     * Any normal return is success.  Any exception thrown is failure.
+     */
+    private fun testRest(
+        restTransportType: RESTTransportType,
+        receiver: Receiver,
+        responseBody: MutableList<String>
+    ): Boolean {
+        logger.info("REST Transport $restTransportType")
+        responseBody.add("${receiver.fullName}: REST Transport")
         try {
-            responseBody.add("**** Receiver Status: ${receiver.customerStatus}")
-            return when (receiver.transport) {
-                is SFTPTransportType -> {
-                    testSftp(receiver.transport, receiver, sftpFile, responseBody)
-                    responseBody.add("**** ${receiver.fullName}: OK")
-                    true
-                }
-                // todo add other types of transports as needed.
-                else -> {
-                    responseBody.add(
-                        "**** ${receiver.fullName}: No test implemented for transport type ${receiver.transport.type}"
-                    )
-                    false
+            val theRESTTransport = RESTTransport()
+            val reportId = UUID.randomUUID().toString()
+            // REST transport throws exception with error method to handle fails
+            // get the username/password to authenticate with OAuth, fail throws exception
+            val credential: RestCredential = theRESTTransport.lookupDefaultCredential(receiver)
+
+            // get the TLS/SSL cert in a JKS if needed, NY uses a specific one, fail throws exception
+            val jksCredential = restTransportType.tlsKeystore?.let { theRESTTransport.lookupJksCredentials(it) }
+
+            responseBody.add("Attempting to authenticate at: ${restTransportType.authTokenUrl}")
+            val aLogger: Logger = Logger.getLogger(this.toString())
+            runBlocking {
```

Comment:
nice.
Did we test the time-out code flow here? It's likely to actually happen in production?